pkg/agent/openflow/client_test.go

```diff
@@ -152,58 +151,42 @@ func TestConcurrentFlowInstallation(t *testing.T) {
 			client := ofClient.(*client)
 			client.flowOperations = m
 
-			var wg sync.WaitGroup
-			var counter int32        // tracks the number of concurrent Add calls at any given time
-			var callIdx int32        // used to distinguish between the first Add call and subsequent calls
-			var concurrentCalls bool // set to true if we observe concurrent calls
-			var stop bool
-			var mutex sync.Mutex
-			cond := sync.NewCond(&mutex)
+			var concurrentCalls atomic.Bool // set to true if we observe concurrent calls
 
+			addCh := make(chan string)
 			m.EXPECT().Add(gomock.Any()).DoAndReturn(func(args ...interface{}) error {
-				cond.L.Lock()
-				defer cond.L.Unlock()
-				counter += 1
-				callIdx += 1
-				if callIdx == 1 { // first call
-					// we wait until we need to stop
-					// we stop when one of these conditions is met:
-					// 1) we detected concurrent Add calls (test is a success), or
-					// 2) the test timed out
-					for !stop {
-						cond.Wait()
-					}
-				} else if counter > 1 {
-					concurrentCalls = true
-					stop = true
-					cond.Signal()
+				x := uuid.New()
+				select {
+				case <-addCh:
+					concurrentCalls.Store(true)
```

Comment:
Could it use "sync/atomic" to avoid introducing new dependency?
```
import "sync/atomic"
concurrentCalls := &atomic.Value{}
concurrentCalls.Store(true)
```

Reply:
Sure, updated.